### ISSUE TITLE
Suggestion: hidden list field with current item index

### ISSUE THREAD

**Issue by josineto** [sveltia/sveltia-cms#172]
The idea is: in a List widget with fields (e.g. not "simple string list"), one could add a hidden field that holds the index of the current item in that list, like:

```yaml
list:
- fruit: apple
  index: 0 # <---
- fruit: banana
  index: 1 # <---
```

It could be a `hidden` widget with `default: '{{index}}'`, or a `compute` widget with `value: '{{index}}'`, or even something else...

**Comment by kyoshino** [comment]
I could add support for the `index`, but indexes can be easily retrieved in the framework, right? In Svelte, that’s the `each` block’s second argument: https://svelte.dev/docs/logic-blocks#each

**Comment by josineto** [comment]
> indexes can be easily retrieved in the framework

Not always: in Hugo, it's idiomatic to filter a collection right before looping into it (a `range (where <collection> <filter>` expression. In that case, the indexes are not relative to the original collection, but to the filtered one, so an `index` field would be useful.

**Comment by kyoshino** [comment]
Okay, let’s see how the hidden `index` field would work.

**Comment by kyoshino** [comment]
The fix shipped with [v0.35.0](https://github.com/sveltia/sveltia-cms/releases/tag/v0.35.0) 🚢

**Comment by josineto** [comment]
Thank you!
